android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -258,10 +262,13 @@ fun purchaseSubscriptionOption(
             // Iterates over StoreProducts and SubscriptionOptions to find
             // the first matching product id and subscription option id
             val optionToPurchase = storeProducts.firstNotNullOfOrNull { storeProduct ->
-                storeProduct.subscriptionOptions?.firstOrNull { subscriptionOption ->
-                    storeProduct.purchasingData.productId == productIdentifier &&
+                // Create StoreProduct copy with presentedOfferingIdentifier if exists
+                // This will give all SubscriptionOption the presentedOfferingIdentifier
+                storeProduct.applyOfferingIdentifier(presentedOfferingIdentifier)
+                    .subscriptionOptions?.firstOrNull { subscriptionOption ->
+                        storeProduct.purchasingData.productId == productIdentifier &&
                             subscriptionOption.id == optionIdentifier
-                }
```

Comment:
Applying the offering identifier to a `GoogleStoreProduct` will also apply it to all the subscription options.

This is kind of an inefficient way to do this but we don't expose a nice way to applying the offering identifier on a `SubscriptionOption` without having to cast to a `GoogleSubscriptionOption` which feels bad.

The cleanest way to do it is to just apply it a `StoreProduct` level which a `GoogleStoreProduct` will then apply it to all of its `SubscriptionOptions`.

This could be something to improve later on.